core/templates/pages/exploration-player-page/services/exploration-player-state.service.ts

```diff
@@ -84,7 +84,9 @@ export class ExplorationPlayerStateService {
     ReadOnlyExplorationBackendApiService,
     private statsReportingService: StatsReportingService,
     private urlService: UrlService
-  ) {
+  ) {}
+
+  ngOnInit(): void {
```

Comment:
Moved constructor logic to ngOnInit.
Since In angular primary purpose of a constructor is Dependency Injection.
It's hard to test the code in the constructor with different edge case scenarios Since Unit tests automatically call the constructor (unlike ngOninit) we cant add pre-checks.
Reference: https://stackoverflow.com/a/48705981/16205538